hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotDefragAdmin.java

```diff
@@ -0,0 +1,221 @@
+    }
+  }
+
+  /**
+   * Tests triggering snapshot defrag with --no-wait option.
+   */
+  @Test
+  public void testDefragWithNoWait() throws Exception {
+    OzoneManager leader = cluster.getOMLeader();
+    String leaderId = leader.getOMNodeId();
+
+    executeDefragCommand(leaderId, true);
+  }
+
+  /**
+   * Tests triggering snapshot defrag on a follower with --no-wait option.
+   */
+  @Test
+  public void testDefragOnFollowerWithNoWait() throws Exception {
+    OzoneManager leader = cluster.getOMLeader();
+    List<OzoneManager> allOMs = cluster.getOzoneManagersList();
+
+    // Find a follower OM
+    OzoneManager follower = null;
+    for (OzoneManager om : allOMs) {
+      if (!om.getOMNodeId().equals(leader.getOMNodeId())) {
+        follower = om;
+        break;
+      }
+    }
+
+    assertNotNull(follower, "Should have at least one follower OM");
+    executeDefragCommand(follower.getOMNodeId(), true);
+  }
+
+  /**
+   * Helper method to execute the defrag command on a specific OM node.
+   *
+   * @param nodeId the OM node ID to target
+   * @param noWait whether to use the --no-wait option
+   */
+  private void executeDefragCommand(String nodeId, boolean noWait) throws Exception {
+    OzoneAdmin ozoneAdmin = new OzoneAdmin();
+    ozoneAdmin.getOzoneConf().addResource(cluster.getConf());
+
+    // Capture output to verify command execution
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream ps = new PrintStream(baos, true, StandardCharsets.UTF_8.name());
+    PrintStream oldOut = System.out;
+    System.setOut(ps);
+
+    try {
+      String[] args;
+      if (noWait) {
+        args = new String[]{
+            "om",
+            "snapshot",
+            "defrag",
+            "-id", omServiceId,
+            "--node-id", nodeId,
+            "--no-wait"
+        };
+      } else {
+        args = new String[]{
+            "om",
+            "snapshot",
+            "defrag",
+            "-id", omServiceId,
+            "--node-id", nodeId
+        };
+      }
+
+      int exitCode = ozoneAdmin.execute(args);
+      System.out.flush();
+      String output = baos.toString(StandardCharsets.UTF_8.name());
+
+      // Verify successful execution
+      assertEquals(0, exitCode,
+          "Command should execute successfully on OM " + nodeId);
+      assertTrue(output.contains("Triggering Snapshot Defrag Service"),
+          "Output should indicate defrag service is being triggered");
+
+      if (noWait) {
+        assertTrue(output.contains("triggered successfully") &&
+                output.contains("background"),
+            "Output should indicate task triggered in background: " + output);
+      } else {
+        assertTrue(output.contains("completed successfully") ||
+                output.contains("failed") ||
+                output.contains("interrupted"),
+            "Output should indicate completion status: " + output);
+      }
+    } finally {
+      System.setOut(oldOut);
```

Comment:
close the PrintStream object ps here.

```suggestion
      System.setOut(oldOut);
      ps.close();
```